lib/request-wrapper.ts

```diff
@@ -580,3 +609,8 @@ function ensureJSONResponseBodyIsObject(response: any): any | string {
 
   return dataAsObject;
 }
+
+// Returns true iff safe headers should be copied to a redirected request.
+function shouldCopySafeHeadersOnRedirect(fromHost: string, toHost: string): boolean {
+  return fromHost.endsWith('.cloud.ibm.com') && toHost.endsWith('.cloud.ibm.com');
```

Comment:
This logic should work as long as we are also supporting the scenario where a redirection to the same host also results in the safe headers being copied, which I assume is the case since you're not completely overriding axios' redirection behavior (presumably) :)

Reply:
Those scenarios are covered by the unit tests now and yes, we are not overriding the whole redirection behavior, it's just a sort of post-processing in the chain, before sending the next, redirected request.